### ISSUE TITLE
Joint Entropy is not computed correctly

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/metalearn#145]
@mitar showed me this example, demonstrating that our method of [computing joint entropy](https://github.com/byu-dml/metalearn/blob/develop/metalearn/metafeatures/information_theoretic_metafeatures.py#L19) is wrong.

```
>>> a = pandas.Series([11,11,11])
>>> b = pandas.Series([22,22,22])
>>> a.astype(str) + b.astype(str)
0    1122
1    1122
2    1122
dtype: object
```
joint entropy is 0 and we would compute 0
```
>>> a = pandas.Series([1,11,112])
>>> b = pandas.Series([122,22,2])
>>> a.astype(str) + b.astype(str)
0    1122
1    1122
2    1122
dtype: object
```
joint entropy is 1.0986... but we would also compute 0.

**Comment by bjschoenfeld** [comment]
I think pairing the elements together in a tuple or even placing a token between the strings (like a space) would solve the problem.

**Comment by mitar** [comment]
Why are you even computing it this way? Why not do a normal joint entropy computation?

**Comment by bjschoenfeld** [comment]
Combining the columns in this way (or rather the correct implementation of this idea :)) still gives us the joint probability distribution (where it is non-zero). It also allows us to reuse the same `entropy` function. Links to a normal computation are welcome though.

A special token would not work, in the case that the columns are sequences of that special token. Tuples would still work.

**Comment by mitar** [comment]
So I just implemented my "reference" [metafeatures computation primitive](https://gitlab.com/datadrivendiscovery/common-primitives/merge_requests/10/diffs). Feedback would be greatly appreciated. I will also be making a MR with some proposed changes to metafeatures schema. I found some things which could be greatly simplified and generalized. (The primitive is already doing it according to it).

I would love if you compare numbers and let's see where are differences.

Also, code is much uglier than yours and much less modular.

[Here is me computing joint entropy](https://gitlab.com/datadrivendiscovery/common-primitives/merge_requests/10/diffs#f6a059ce2e83058a3a41cb90c90c3e6a313b7845_0_398). In short, I compute probabilities and then I compute entropy. It is pretty simple so I am not sure if it is relevant to try to reuse entropy function. [See here for more information about this approach](https://blog.biolab.si/2012/06/15/computing-joint-entropy-in-python/) (which is accidentally a blog of my lab where I did my undergrad research).
